src/Component/JsonSchema/Jane.php

```diff
@@ -58,24 +58,22 @@ public function createContext(Registry $registry): Context
         foreach ($registry->getSchemas() as $schema) {
             foreach ($schema->getClasses() as $class) {
                 $properties = $this->chainGuesser->guessProperties($class->getObject(), $schema->getRootName(), $class->getReference(), $registry);
+                // map: property name => count
                 $names = [];
 
                 foreach ($properties as $property) {
-                    $property->setPhpName($this->naming->getPropertyName($property->getName()));
-
-                    $i = 2;
-                    $newName = $property->getPhpName();
-
-                    while (\in_array(strtolower($newName), $names, true)) {
-                        $newName = $property->getPhpName() . $i;
-                        ++$i;
-                    }
-
-                    if ($newName !== $property->getPhpName()) {
-                        $property->setPhpName($newName);
+                    $cleanedName = $this->naming->cleaning($property->getName());
+                    $duplicateName = strtolower($cleanedName);
```

Comment:
I would prefer to have this logic in the Naming class

Reply:
I was not sure about what part you want in the Naming class. I move all the de-duplication logic in it.
I hope I don't misunderstood you =/